### DISCUSSION TITLE
Update SimpleIdServer.Scim from version 4.0.0 on 4.0.5

### DISCUSSION THREAD

**Discussion by alexander-durovich-swi** [simpleidserver/SimpleIdServer#638] [Q&A]
Hello,

We are using your SimpleIdServer.Scim project and we are trying to update from version 4.0.0 on 4.0.5.

We see the following changes by working with ISCIMRepresentationCommandRepository:
Now by calling the ISCIMRepresentationCommandRepository.Update method from https://github.com/simpleidserver/SimpleIdServer/blob/v4.0.5/src/Scim/SimpleIdServer.Scim/Commands/Handlers/PatchRepresentationCommandHandler.cs#L79 the passed existingRepresentation argument has FlatAttributes with old data (i.e. not updated with new data from http request). Earlier the FlatAttributes had updated values, and we converted existingRepresentation into out model in the Update method.

Could you say if it is the supposed behaviour now or bug, please?
And if it is supposed behaviour now, do we understand right, that now attributes with new data can be processed only by calling BulkXX methods from
https://github.com/simpleidserver/SimpleIdServer/blob/v4.0.5/src/Scim/SimpleIdServer.Scim/Commands/Handlers/PatchRepresentationCommandHandler.cs#L69-L71?

Thank you in advance.
Best Regards,
Alex

**Comment by simpleidserver** [comment]
Hello,

There has been a change in the method used by the library to update the SCIM Representation since version `4.0.4`.

While working on very large SCIM representations, such as a Group with one million users, I observed performance issues:
* The change tracking of Entity Framework consumed excessive memory due to tracking all attributes of the representation (Change tracking).
* Loading the entire SCIM representation into memory proved to be problematic.

Instead of loading the entire representation into memory, I now compute the difference between the old and new representation. 
I extract the result into a list of PATCH operations and execute Bulk operations to update the attributes. 
This method is more efficient and does not result in memory leaks.

If you require assistance with your migration, I am available to help.

Kind regards,

SID

**Comment by alexander-durovich-swi** [comment]
Hello,

I see, thank you very nuch for your answer and your work!

Best Regards,
Alex
